contracts/eip721/EIP721.sol

```diff
@@ -0,0 +1,311 @@
+pragma solidity ^0.4.21;
+import "./EIP721Interface.sol";
+import "./EIP721MetadataInterface.sol";
+import "./EIP721EnumerableInterface.sol";
+import "./EIP721TokenReceiverInterface.sol";
+
+/*
+This is a full implementation of all ERC721's features.
+Influenced by OpenZeppelin's implementation with some stylistic changes.
+https://github.com/OpenZeppelin/zeppelin-solidity/tree/master/contracts/token/ERC721
+*/
+
+contract EIP721 is EIP721Interface, EIP721MetadataInterface, EIP721EnumerableInterface, ERC165Interface {
+    string public name;
+    string public symbol;
+
+    // all tokens
+    uint256[] internal allTokens;
+    mapping(uint256 => uint256) internal allTokensIndex;
+    // Array of tokens owned by a specific owner
+    mapping(address => uint256[]) internal ownedTokens;
+    // Mapping from token ID to owner
+    mapping(uint256 => address) internal ownerOfToken;
+    // Mapping of the token ID to where it is in the owner's array.
+    mapping(uint256 => uint256) internal ownedTokensIndex;
+
+    // Mapping of a token to a specifically approved owner.
+    mapping(uint256 => address) internal approvedOwnerOfToken;
+
+    // An operator is allowed to manage all assets of another owner.
+    mapping(address => mapping (address => bool)) internal approvedOperators;
```

Comment:
Could this just be called operators? It technically contains all operators, just with their boolean value flag set to false. Like, if I have some operator who I approve, and then I revoke the approval, we make the revocation assignment to the _approved_Operators mapping, which seems semantically questionable.